### DISCUSSION TITLE
Can MinIO be used as a Sentry Native Symbol Server?

### DISCUSSION THREAD

**Discussion by gdianaty** [getsentry/support#14] [Support] [answered]
As it stands, the GUI for Sentry does not allow the user to set their own endpoint when configuring an S3 Debug Information Files Repository.

Without the ability to specify an endpoint for the S3 server, an alternative to S3 such as MinIO cannot be used to store PDB files for use with Sentry Native.

Am I wrong? Is it possible to use an S3 alternative such as MinIO to store PDB files for Sentry?

![image](https://user-images.githubusercontent.com/9407540/171983663-e22acab1-8e0f-4436-8884-3ff824ede9b7.png)

**Comment by mq1n** [comment]
is there any update about that?

**Comment by souredoutlook** [comment, marked as answer]
Hey @mq1n @gdianaty  - I've escalated this question to the Ingest team. Thanks for your patience.

**Reply by souredoutlook** [reply]
Hey @mq1n @gdianaty - I've got an update from the team:

>Amazon S3 Repositories are only targeted at Amazon provided Fileservices. Although MinIO is a filestore with an S3 compatible interface that you can host yourself, as we do not support adding IP address and the like to target such a filestore, MinIO cannot be used as of today, even it would be compatible with an S3 protocol.

**Reply by mq1n** [reply]
MinIO is not using IP address, it's uses HTTP protocol just like raw HTTP symbol server, I don't see any reason to worry about place some host address.

**Reply by souredoutlook** [reply]
I'll pass that back to the team but I believe the point still stands that it is not currently possible to use MinIO as a symbol server with Sentry.

**Reply by gdianaty** [reply]
Actually, we explored simply modifying Sentry to support such a behavior. The engineering team that looked into it informed me it may be fairly easy.

Much of the existing S3 code can be seamlessly recycled, because, as @mq1n put it, it's a protocol. In fact, Amazon's own existing APIs for the use of their hosted file services are compatible with MinIO, by design. That includes Sentry! It's very cool stuff.

All that's really required would be a GUI for punching the information in so you direct the existing S3 integration to an instance of MinIO.
